InformationScripting/src/parsing/ArgumentParser.cpp

```diff
@@ -41,21 +42,27 @@ ScopedArgumentQuery::ScopedArgumentQuery(Model::Node* target, std::initializer_l
 	argParser_->setSingleDashWordOptionMode(QCommandLineParser::ParseAsLongOptions);
 
 	if (!argParser_->parse(args))
-		qWarning() << args[0] << "parse failure"; // TODO warn user
+		throw ParsingException("General parsing error");
```

Comment:
We should probably call this something a bit more descriptive, e.g. `QueryParsingException`.